warrior/Framework/Utils/cli_Utils.py

```diff
@@ -595,6 +595,24 @@ def send_commands_from_testdata(testdatafile, obj_session, **args):
                                                           result=result, response=response, system_name=system_name)
                 response_dict = _get_response_dict(details_dict, i, response,
                                                    response_dict)
+                sys_name = ses_name = ''
+                if response_dict.values()[i] is not None:
+                    if details_dict["sys_list"][i] is '' or details_dict["sys_list"][i] is None:
+                        sys_name = system_name
+                    else:
+                        sys_name = details_dict["sys_list"][i]
+                    if details_dict["session_list"][i] is '' or details_dict["session_list"][i] is None:
+                        session_name = args.get("session_name")
+                        ses_name = session_name
+                    else:
+                        ses_name = details_dict["session_list"][i]
+
+                    session_id = data_Utils.get_session_id(sys_name, ses_name) + "_td_response"
+                    for k, v in response_dict.items():
+                        if k is response_dict.keys()[i]:
```

Comment:
1. Python dictionary is unordered, using index position might go wrong sometimes.
2. Use "==" for comparing values.

Reply:
Addressed your comments.